src/executePrettier/executePrettierOnBufferRange.js

```diff
@@ -36,7 +36,13 @@ const executePrettierOnBufferRange = (editor: TextEditor, bufferRange: Range) =>
     return;
   }
 
-  editor.setTextInBufferRange(bufferRange, transformed);
+  const editorBuffer = editor.getBuffer();
+  if (editorBuffer.getRange().isEqual(bufferRange)) {
+    editorBuffer.setTextViaDiff(transformed);
+  } else {
+    editor.setTextInBufferRange(bufferRange, transformed);
+  }
```

Comment:
j/c, why this check? Why not always use `setTextViaDiff`? In case the user changed something after we started formatting?

Reply:
It's needed to support formatting only selected code when the **Format** command is invoked manually (not on save), since `setTextViaDiff` doesn't take a range parameter.

Reply:
Can you help with test case? I've never used Flow or Jest. Here's what I have so far:

```js
it('sets the transformed text via diff when buffer equals entire range of editor', () => {
  editor.getBuffer.mockImplementation(() => ({
    getRange: () => ({ isEqual: () => true }),
    setTextViaDiff: () => bufferRangeFixture,
  }));

  executePrettierOnBufferRange(editor, bufferRangeFixture);

  expect(prettier.format).toHaveBeenCalledWith('const foo = (2);', { useTabs: false });
  // expect(editor.getBuffer.setTextViaDiff).toHaveBeenCalledWith('const foo = 2;');
});
```

But I'm getting the following error:

```sh
sets the transformed text via diff when buffer equals entire range of editor

   expect(jest.fn())[.not].toHaveBeenCalledWith()

   jest.fn() value must be a mock function or spy.
   Received: undefined

     at Object.<anonymous> (src/executePrettier/executePrettierOnBufferRange.test.js:48:43)
         at new Promise (<anonymous>)
         at <anonymous>
     at process._tickCallback (internal/process/next_tick.js:188:7)
```

Any advice? Thanks.

Reply:
> It's needed to support formatting only selected code when the Format command is invoked manually (not on save), since setTextViaDiff doesn't take a range parameter.

Oh, duh! Still, maybe we should put in a var or something just to make it clear why you're doing what you're doing there?

As far as that code snippet, the problem is you're accessing `editor.getBuffer.setTextViaDiff`, but `getBuffer` is a function not an object. Something like this:

```js
  const setTextViaDiffMock = jest.fn();
  editor.getBuffer.mockImplementation(() => ({
    getRange: () => ({ isEqual: () => true }),
    setTextViaDiff: setTextViaDiffMock,
  }));

  executePrettierOnBufferRange(editor, bufferRangeFixture);

  expect(prettier.format).toHaveBeenCalledWith('const foo = (2);', { useTabs: false });
  expect(setTextViaDiffMock).toHaveBeenCalledWith('const foo = 2;');
```

Reply:
Your test code worked perfectly – thanks so much. I'll issue a separate PR with it shortly.

Reply:
> maybe we should put in a var or something just to make it clear why you're doing what you're doing there?

What if I added a comment that explains the rationale? That would allow a little more exposition.